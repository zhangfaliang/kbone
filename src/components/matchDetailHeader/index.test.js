import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import MatchDetailHeater from './index';

describe('MatchDetailHeater', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<MatchDetailHeater />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('centerText', () => {
    expect.assertions(1);
    const wrapper = render(<MatchDetailHeater centerText="sdfjsd" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('shouldComponentUpdate activeRightIconName', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailHeater activeRightIconName={false} />);
    wrapper.setProps({
      activeRightIconName: true,
    });
    expect(wrapper.state('activeRightIconName')).toBe(true);
  });
  it('handleRightIcn-vs', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailHeater showVSIcon={true} />);
    const handelVSIcon = wrapper.prop('handelVSIcon');
    const click = jest.fn(handelVSIcon);
    wrapper
      .find('MatchDetailHeater')
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .children('span')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handelVSIcon');
  });
  it('handleRightIcn-HappeningIcon', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailHeater showVSIcon={true} showHappeningIcon={true} />);
    const handelHappeningIcon = wrapper.prop('handelHappeningIcon');
    const click = jest.fn(handelHappeningIcon);
    wrapper
      .find('MatchDetailHeater')
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .children('span')
      .at(1)
      .simulate('click');
    expect(click()).toBe('handelHappeningIcon');
  });
  it('handelBackIcon', () => {
    expect.assertions(1);
    const wrapper = mount(<MatchDetailHeater showVSIcon={true} showHappeningIcon={true} />);
    const handelBackIcon = wrapper.prop('handelBackIcon');
    const click = jest.fn(handelBackIcon);
    wrapper
      .find('MatchDetailHeater')
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .children('span')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handelBackIcon');
  });
});
