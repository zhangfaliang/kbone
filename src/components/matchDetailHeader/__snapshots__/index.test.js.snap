// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchDetailHeater base 1`] = `
<div
  class="default-match-detail-header"
>
  <div
    class="left"
  >
    <span
      class="leftIcon"
    >
      <span
        class="iconfont icon-mycopy"
      />
    </span>
    <span
      class="leftText"
    >
      NBA
    </span>
  </div>
  <div
    class="right"
  >
    <span
      class="rigthIcon activeIcon"
    >
      <span
        class="iconfont icon-bisaixinxi"
      />
    </span>
    <span
      class="rigthIcon"
    >
      <span
        class="iconfont icon-shikuangxinxicopy"
      />
    </span>
  </div>
</div>
`;

exports[`MatchDetailHeater centerText 1`] = `
<div
  class="default-match-detail-header"
>
  <div
    class="left"
  >
    <span
      class="leftIcon"
    >
      <span
        class="iconfont icon-mycopy"
      />
    </span>
    <span
      class="leftText"
    >
      NBA
    </span>
  </div>
  <span
    class="center"
  >
    sdfjsd
  </span>
  <div
    class="right"
  >
    <span
      class="rigthIcon activeIcon"
    >
      <span
        class="iconfont icon-bisaixinxi"
      />
    </span>
    <span
      class="rigthIcon"
    >
      <span
        class="iconfont icon-shikuangxinxicopy"
      />
    </span>
  </div>
</div>
`;
